src/portable/st/stm32f4/dcd_stm32f4.c

```diff
@@ -80,20 +80,39 @@ static void bus_reset(void) {
   dev->DOEPMSK |= USB_OTG_DOEPMSK_STUPM | USB_OTG_DOEPMSK_XFRCM;
   dev->DIEPMSK |= USB_OTG_DIEPMSK_TOM | USB_OTG_DIEPMSK_XFRCM;
 
+  // Peripheral FIFO architecture (Rev18 RM 29.11)
```

Comment:
@cr1901 please let me know if I understand the Reference Manual's FIFO allocation correctly. All RM seem to be written by hw engineer who expect everyone is an expect of their design, so it is easy to mis-interprete :D.

Reply:
@hathach I added my own comments to my interpretation of the FIFO allocation described in the RM. Enjoy! :)



>`// * 2 locations for OUT endpoint control words.`

One control word is sent when the OUT packet is received. The other control word is when the OUT packet has been fully read from the FIFO.